sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -200,6 +200,8 @@ object RemoveAliasOnlyProject extends Rule[LogicalPlan] {
         case plan: Project if plan eq proj => plan.child
         case plan => plan transformExpressions {
           case a: Attribute if attrMap.contains(a) => attrMap(a)
+          case b: Alias if attrMap.exists(_._1.exprId == b.exprId)
+            && b.child.isInstanceOf[NamedExpression] => b.child
```

Comment:
As you said, if we find an alias-only project, e.g. alias a#1 to a#2, it's safe to `remove this project` and `replace all a#2 with a#1` in this plan. So another Alias which is also alias a#1 to a#2, but not the same object with the first one, it will not be processed. 

here, the logic shows that we process the situation.